src/views/v3/Bridge/Routes/index.tsx

```diff
@@ -62,46 +64,74 @@ function Routes({
 
   // Event handlers
   const handleCloseModal = useCallback(() => {
-    setHighlightedRoute(selectedRoute);
     setShowModal(false);
-  }, [selectedRoute]);
+    // Only revert if route selection changed (not just gas amount on same route)
+    if (highlightedRoute !== originalSelectedRoute) {
+      setHighlightedRoute(originalSelectedRoute);
+      dispatch(setToNativeToken(originalToNativeToken));
+    }
+  }, [
+    highlightedRoute,
+    originalSelectedRoute,
+    originalToNativeToken,
+    dispatch,
+  ]);
 
   const handleCloseDrawer = useCallback(() => {
     setShowDrawer(false);
-  }, []);
+    // Only revert if route selection changed (not just gas amount on same route)
+    if (highlightedRoute !== originalSelectedRoute) {
+      setHighlightedRoute(originalSelectedRoute);
+      dispatch(setToNativeToken(originalToNativeToken));
+    }
+  }, [
+    highlightedRoute,
+    originalSelectedRoute,
+    originalToNativeToken,
+    dispatch,
+  ]);
 
   const handleToggleRoutes = useCallback(() => {
+    // Store the original values when opening modal/drawer
+    setOriginalToNativeToken(toNativeToken);
+    setOriginalSelectedRoute(selectedRoute);
     if (mobile) {
       setShowDrawer(true);
     } else {
       setShowModal((prev) => !prev);
     }
-  }, [mobile]);
+  }, [mobile, selectedRoute, toNativeToken]);
 
-  const handleRouteSelect = useCallback((route: string) => {
-    setHighlightedRoute(route);
-  }, []);
+  const handleRouteSelect = useCallback(
+    (route: string) => {
+      setHighlightedRoute(route);
+      if (toNativeToken !== 0) {
+        dispatch(setToNativeToken(0));
+      }
+    },
+    [dispatch, toNativeToken],
+  );
 
-  const handleGasTokenChange = useCallback((value: number) => {
-    setGasTokenAmount(value);
-  }, []);
+  const handleGasTokenChange = useCallback(
+    (value: number) => {
+      if (value !== toNativeToken) {
+        dispatch(setToNativeToken(value));
+      }
+    },
+    [dispatch, toNativeToken],
+  );
 
-  const handleSelectRoute = useCallback(() => {
+  const handleRouteConfirm = useCallback(() => {
     if (highlightedRoute) {
       onRouteChange(highlightedRoute);
     }
-    if (gasTokenAmount !== toNativeToken) {
-      dispatch(setToNativeToken(gasTokenAmount));
-    }
+
+    // Update the original values as well since user confirmed their selection
+    setOriginalToNativeToken(toNativeToken);
+    setOriginalSelectedRoute(highlightedRoute);
+
     mobile ? setShowDrawer(false) : setShowModal(false);
```

Comment:
Similar comment as above, you can choose to not have this conditional check, just set both to false.